shared/wallets/transaction-details/index.js

```diff
@@ -431,12 +431,13 @@ class LoadTransactionDetails extends React.Component<Props> {
   }
   componentDidUpdate(prevProps: Props) {
     // An erased transaction ID likely means the payment was updated,
+    // which means details need to be retrieved again.
+    // Also, recentPaymentsReceived can race with paymentDetailReceived
+    // so that loading finishes before transactionID is ever seen here.
     // which means details need to be retrieved again
     if (
-      (!this.props.transactionID || !this.props.senderAccountID) &&
-      prevProps.transactionID &&
-      prevProps.senderAccountID &&
-      !this.props.loading
+      (!this.props.transactionID && prevProps.transactionID) ||
```

Comment:
Should this one also `&& !this.props.loading`?

Reply:
prettier did that! I added parens though and it seems ok now